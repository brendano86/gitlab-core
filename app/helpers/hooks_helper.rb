# frozen_string_literal: true

module HooksHelper
  def link_to_hook_docs(hook)
    case hook
    when SystemHook
      link_to _('System hooks'), help_page_path('system_hooks/system_hooks')
    else
      link_to _('Webhooks'), help_page_path('user/project/integrations/webhooks')
    end
  end

  def link_to_test_hook(hook, trigger)
    path = case hook
           when GroupHook
             test_group_hook_path(hook.group, hook, trigger: trigger)
           when ProjectHook
             project = hook.project
             test_project_hook_path(project, hook, trigger: trigger)
           when SystemHook
             test_admin_hook_path(hook, trigger: trigger)
           end

    trigger_human_name = trigger.to_s.tr('_', ' ').camelize

    link_to path, rel: 'nofollow', method: :post do
      content_tag(:span, trigger_human_name)
    end
  end

  def link_to_edit_hook(hook)
    case hook
    when GroupHook
      edit_group_hook_path(hook.group, hook)
    when ProjectHook
      edit_project_hook_path(hook.project, hook)
    when SystemHook
      edit_admin_hook_path(hook)
    end
  end

  def link_to_destroy_hook(hook)
    case hook
    when GroupHook
      group_hook_path(hook.group, hook)
    when ProjectHook
      project_hook_path(hook.project, hook)
    when SystemHook
      admin_hook_path(hook)
    end
  end
end
