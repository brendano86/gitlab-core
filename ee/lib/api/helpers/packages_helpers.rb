# frozen_string_literal: true

module API
  module Helpers
    module PackagesHelpers
      def require_packages_enabled!
        not_found! unless ::Gitlab.config.packages.enabled
      end

      def authorize_packages_access!(subject)
        require_packages_enabled!
        authorize_packages_feature!(subject)
        authorize_read_package!(subject)
      end

      def authorize_packages_feature!(subject)
        forbidden! unless subject.feature_available?(:packages)
      end

      def authorize_read_package!(subject)
        authorize!(:read_package, subject)
      end

      def authorize_create_package!(subject)
        authorize!(:create_package, subject)
      end

      def authorize_destroy_package!(subject)
        authorize!(:destroy_package, subject)
      end
    end
  end
end
