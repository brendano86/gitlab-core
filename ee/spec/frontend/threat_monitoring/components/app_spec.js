import { createLocalVue, shallowMount } from '@vue/test-utils';
import { GlAlert } from '@gitlab/ui';
import { TEST_HOST } from 'helpers/test_constants';
import createStore from 'ee/threat_monitoring/store';
import ThreatMonitoringApp from 'ee/threat_monitoring/components/app.vue';

const localVue = createLocalVue();
const endpoint = TEST_HOST;
const emptyStateSvgPath = '/svgs';
const documentationPath = '/docs';

describe('ThreatMonitoringApp component', () => {
  let store;
  let wrapper;

  const factory = propsData => {
    store = createStore();

    jest.spyOn(store, 'dispatch');

    wrapper = shallowMount(ThreatMonitoringApp, {
      localVue,
      propsData,
      store,
      sync: false,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('given the WAF is not set up', () => {
    beforeEach(() => {
      factory({
        isWafSetup: false,
        endpoint,
        emptyStateSvgPath,
        documentationPath,
      });
    });

    it('does not dispatch any store actions', () => {
      expect(store.dispatch).not.toHaveBeenCalled();
    });

    it('shows only the empty state', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('given the WAF is set up', () => {
    beforeEach(() => {
      factory({
        isWafSetup: true,
        endpoint,
        emptyStateSvgPath,
        documentationPath,
      });
    });

    it('sets the endpoint on creation', () => {
      expect(store.dispatch).toHaveBeenCalledWith('threatMonitoring/setEndpoint', endpoint);
    });

    it('shows the alert and header', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('dismissing the alert', () => {
      beforeEach(() => {
        wrapper.find(GlAlert).vm.$emit('dismiss');
      });

      it('hides the alert', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
