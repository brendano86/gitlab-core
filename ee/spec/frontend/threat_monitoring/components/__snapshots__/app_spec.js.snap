// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreatMonitoringApp component given the WAF is not set up shows only the empty state 1`] = `
<glemptystate-stub
  description="A Web Application Firewall (WAF) provides monitoring and rules to protect production applications. GitLab adds the modsecurity WAF plug-in when you install the Ingress app in your Kubernetes cluster."
  primarybuttonlink="/docs"
  primarybuttontext="Learn more"
  svgpath="/svgs"
  title="Web Application Firewall not enabled"
/>
`;

exports[`ThreatMonitoringApp component given the WAF is set up dismissing the alert hides the alert 1`] = `
<section>
  <!---->
   
  <header
    class="my-3"
  >
    <h2
      class="h4 mb-1"
    >
      
      Threat Monitoring
      
      <gllink-stub
        aria-label="Threat Monitoring help page link"
        href="/docs"
        target="_blank"
      >
        <glicon-stub
          name="question"
          size="16"
        />
      </gllink-stub>
    </h2>
  </header>
</section>
`;

exports[`ThreatMonitoringApp component given the WAF is set up shows the alert and header 1`] = `
<section>
  <glalert-stub
    class="my-3"
    dismissible="true"
    dismisslabel="Dismiss"
    primarybuttonlink=""
    primarybuttontext=""
    secondarybuttonlink="/docs"
    secondarybuttontext="View documentation"
    title=""
    variant="tip"
  >
    
    The graph below is an overview of traffic coming to your application. View the documentation for instructions on how to access the WAF logs to see what type of malicious traffic is trying to access your app.
  
  </glalert-stub>
   
  <header
    class="my-3"
  >
    <h2
      class="h4 mb-1"
    >
      
      Threat Monitoring
      
      <gllink-stub
        aria-label="Threat Monitoring help page link"
        href="/docs"
        target="_blank"
      >
        <glicon-stub
          name="question"
          size="16"
        />
      </gllink-stub>
    </h2>
  </header>
</section>
`;
